### ISSUE TITLE
Beautify command line help output

### ISSUE THREAD

**Issue by maettu-this** [oleg-shilo/cs-script#78]
Suggesting to limit the command line help's text width to the console width, in order to prevent line breaks and improve readability.

Some ideas taken from my own [ArgsHandler](https://svn.code.sf.net/p/y-a-terminal/code/trunk/MKY/MKY/CommandLine) making use of [StringEx](https://svn.code.sf.net/p/y-a-terminal/code/trunk/MKY/MKY/Types):

- `Console.Out.Write(GetHelpText(Console.WindowWidth - 1)); // -1 ensures that lines that exactly match the number of characters do not lead to an empty line (due to the NewLine which is added).`
- `string GetHelpText(int maxWidth)`
- `string SplitIntoLines(int maxWidth, int indent, string text)`

```
string SplitIntoLines(int maxWidth, int indent, string text)
{
	StringBuilder lines = new StringBuilder();

	int width = (maxWidth - indent);
	foreach (string line in StringEx.SplitLexically(text, width))
		lines.AppendLine(IndentSpace(indent) + line);

	return (lines.ToString());
}
```
```
string[] SplitLexically(string str, int desiredChunkLength)
{
	List<string> chunks = new List<string>();
	string[] newLineSeparators = new string[] { Environment.NewLine, "\n", "\r" };

	foreach (string paragraph in str.Split(newLineSeparators, StringSplitOptions.None))
		chunks.AddRange(SplitLexicallyWithoutTakingNewLineIntoAccount(paragraph, desiredChunkLength));

	return (chunks.ToArray());
}
```
```
string[] SplitLexicallyWithoutTakingNewLineIntoAccount(string str, int desiredChunkLength)
{
	List<int> spaces = new List<int>();

	// Retrieve all spaces within the string:
	int i = 0;
	while ((i = str.IndexOf(' ', i)) >= 0)
	{
		spaces.Add(i);
		i++;
	}

	// Add an extra space at the end of the string to ensure that the last chunk is split properly:
	spaces.Add(str.Length);

	// Split the string into the desired chunk size taking word boundaries into account:
	int startIndex = 0;
	List<string> chunks = new List<string>();
	while (startIndex < str.Length)
	{
		// Find the furthermost split position:
		int spaceIndex = spaces.FindLastIndex(value => (value <= (startIndex + desiredChunkLength)));
		
		int splitIndex;
		if (spaceIndex >= 0)
			splitIndex = spaces[spaceIndex];
		else
			splitIndex = (startIndex + desiredChunkLength);

		// Limit to split within the string and execute the split:
		splitIndex = Int32Ex.LimitToBounds(splitIndex, startIndex, str.Length);
		int length = (splitIndex - startIndex);
		chunks.Add(str.Substring(startIndex, length));
		startIndex += length;

		// Remove the already used spaces from the collection to ensure those spaces are not used again:
		if (spaceIndex >= 0)
		{
			spaces.RemoveRange(0, (spaceIndex + 1));
			startIndex++; // Advance an extra character to compensate the space.
		}
	}

	return (chunks.ToArray());
}
```

**Comment by oleg-shilo** [comment]
Excellent suggestion. Txs.
Scheduled for the next release

**Comment by oleg-shilo** [comment]
Done.

Thank you for sharing your code. It provided an excellent opportunity to improve CLI support.

I have used a slightly modified version of your `SplitLexicallyWithoutTakingNewLineIntoAccount`. 
I have also introduced a very primitive "markup syntax" to allow sub-paragraphs (sub-indents) so common for CLI help content:

![image](https://user-images.githubusercontent.com/16729806/28609360-b63b5cd4-7227-11e7-9a13-fd6d4a797565.png)

![image](https://user-images.githubusercontent.com/16729806/28609279-6aed97ec-7227-11e7-9d51-0070371f5fcd.png)
